File: csharp/largest-series-product/LargestSeriesProduct.cs
public static class LargestSeriesProduct
{
    public static long GetLargestProduct(string digits, int span) 
    {
        if (span < 0 || span > digits.Length) throw new ArgumentException();
        if (digits.Any(c => !char.IsDigit(c))) throw new ArgumentException();
        if (span == 0) return 1;

        long largestProduct = 0;
        for (int i = 0; i <= digits.Length - span ; i++)
        {
            long currentProduct = 1;
            for (int j = 0; j < span; j++) currentProduct *= (long)char.GetNumericValue(digits[i + j]);
            if (currentProduct> largestProduct) largestProduct = currentProduct;
        }
        return largestProduct;
    }
}

File: javascript/largest-series-product/largest-series-product.js
//
// This is only a SKELETON file for the 'Largest Series Product' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const largestProduct = (digits, span) => {
  if (span < 0) {
    throw new Error('span must not be negative');
  }
  if (/\D/.test(digits)) {
    throw new Error('digits input must only contain digits');
  }
  if (span > digits.length) {
    throw new Error('span must not exceed string length');
  }
  if (span === 0) {
    return 1;
  }
  let maxProduct = 0;
  for (let i = 0; i <= digits.length - span; i++) {
    const window = digits.slice(i, i + span);
    const product = [...window].reduce((acc, num) => acc * Number(num), 1);
    if (product > maxProduct) {
      maxProduct = product;
    }
  }
  return maxProduct;
};
